src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/RedirectHandler.cs

```diff
@@ -54,13 +54,22 @@ public RedirectHandler(int maxAutomaticRedirections, HttpMessageHandler initialI
                 // Clear the authorization header.
                 request.Headers.Authorization = null;
 
+                if (NetEventSource.IsEnabled)
+                {
+                    NetEventSource.Info(this, $"Redirecting from {request.RequestUri} to {redirectUri} in response to status code {(int)response.StatusCode} '{response.StatusCode}'.");
+                }
+
                 // Set up for the redirect
                 request.RequestUri = redirectUri;
                 if (RequestRequiresForceGet(response.StatusCode, request.Method))
                 {
                     request.Method = HttpMethod.Get;
                     request.Content = null;
                     request.Headers.TransferEncodingChunked = false;
+                    if (NetEventSource.IsEnabled)
+                    {
+                        NetEventSource.Info(this, $"Modified request from {request.Method} to GET in response to status code {(int)response.StatusCode} '{response.StatusCode}'.");
```

Comment:
You have lost the original value of request.Method so, this trace won't show it.  It would always show "GET". You need to save the "request.Method" somewhere and use that in the trace.

Also, change literal "GET" (for the updated method) in this string to "{HttpMethod.Get}".

Reply:
Probably better to emit the trace before you actually change request.Method. This way you don't need to allocate another variable just for tracing.

Reply:
Good catch. I can move the logging to before we change the method, which should have less of a performance impact.